src/Audio/AudioOutput.cc

```diff
@@ -7,191 +7,186 @@
  *
  ****************************************************************************/
 
-#include <QDebug>
-#include <QRegularExpression>
-
 #include "AudioOutput.h"
-#include "QGCApplication.h"
-#include "SettingsManager.h"
 
-AudioOutput::AudioOutput(QGCApplication* app, QGCToolbox* toolbox)
-    : QGCTool   (app, toolbox)
-    , _tts      (nullptr)
-{
-    if (qgcApp()->runningUnitTests()) {
-        // Cloud based unit tests don't have speech capabilty. If you try to crank up
-        // speech engine it will pop a qWarning which prevents usage of QT_FATAL_WARNINGS
-        return;
-    }
+#include <QtCore/QRegularExpression>
+
+#define MAX_TEXT_QUEUE_SIZE 20U
+
+Q_LOGGING_CATEGORY( AudioOutputLog, "qgc.audio.audiooutput" );
+// qt.speech.tts.flite
+// qt.speech.tts.android
+
+const QHash<QString, QString> AudioOutput::s_textHash = {
+    { "ERR",            "error" },
+    { "POSCTL",         "Position Control" },
+    { "ALTCTL",         "Altitude Control" },
+    { "AUTO_RTL",       "auto return to launch" },
+    { "RTL",            "return To launch" },
+    { "ACCEL",          "accelerometer" },
+    { "RC_MAP_MODE_SW", "RC mode switch" },
+    { "REJ",            "rejected" },
+    { "WP",             "waypoint" },
+    { "CMD",            "command" },
+    { "COMPID",         "component eye dee" },
+    { "params",         "parameters" },
+    { "id",             "eye dee" },
+    { "ADSB",           "Hey Dee Ess Bee" },
```

Comment:
This one seems a little odd: "Hey" for "A". Isn't that wrong? Might as well fix that.

Reply:
I checked with espeak on linux and I think you can just do "A.D.S.B.", "I.D.", "E.K.F.", etc. I don't know if that's standard though

Reply:
MockLink has a "send status text" option which you could hack into and change the one that is spoken to test it out. I can test it on Mac.

https://github.com/mavlink/qgroundcontrol/blob/master/src/comm/MockLink.cc#L1377

Reply:
I verified that "A.D.S.B" is spoken correctly on the Mac. I had to change the MAV_SEVERITY_NOTICE text in MockLink to test. The "#" status text isn't causing it to talk for some reason. Not sure if there is a bug there, I'll look.